lib/linearizer.js

```diff
@@ -137,11 +139,20 @@ module.exports = class Linearizer {
 
     // get the indexed nodes
     const indexed = []
-    while (true) {
-      const nodes = this.consensus.shift()
-      if (!nodes.length) break
 
-      this._yield(nodes, indexed)
+    // apply consensus heads jit to avoid batch sizes affecting it
+    for (let i = 0; i < this._pendingConsensusHeads.length; i++) {
+      const head = this._pendingConsensusHeads[i]
+      this.consensus.addHead(head)
+      while (true) {
+        const nodes = this.consensus.shift()
+        if (!nodes.length) break
```

Comment:
Might throw of `this.consensus` can be the empty array, with `shift` yielding undefined. Unsure if that's possible, but can be trivially safe with `if (!nodes?.length)

Reply:
Ah, this is not new code, just a different indent, so probably not relevant to change in this PR